dashboard/modules/job/job_manager.py

```diff
@@ -293,36 +339,19 @@ def submit_job(self,
         elif self._status_client.get_status(job_id) is not None:
             raise RuntimeError(f"Job {job_id} already exists.")
 
+        logger.info(f"Starting job with job_id: {job_id}")
         self._status_client.put_status(job_id, JobStatus.PENDING)
 
-        supervisor = None
-        try:
-            logger.debug(
-                f"Submitting job with generated internal job_id: {job_id}")
-            supervisor = self._supervisor_actor_cls.options(
-                lifetime="detached",
-                name=self.JOB_ACTOR_NAME.format(job_id=job_id),
-                num_cpus=0,
-                # Currently we assume JobManager is created by dashboard server
-                # running on headnode, same for job supervisor actors scheduled
-                resources={
-                    self._get_current_node_resource_key(): 0.001,
-                },
-                # For now we assume supervisor actor and driver script have
-                # same runtime_env.
-                runtime_env=runtime_env).remote(job_id, metadata or {})
-            ray.get(supervisor.ready.remote())
-        except Exception as e:
-            if supervisor:
-                ray.kill(supervisor, no_restart=True)
-            self._status_client.put_status(job_id, JobStatus.FAILED)
-            raise RuntimeError(
-                f"Failed to start actor for job {job_id}. This could be "
-                "runtime_env configuration failure or invalid runtime_env."
-                f"Exception message: {str(e)}")
-
-        # Kick off the job to run in the background.
-        supervisor.run.remote(entrypoint, _start_signal_actor)
+        # Wait for the actor to start up in a coroutine so that this call
+        # always returns immediately and we can catch errors with the actor
+        # starting up.
```

Comment:
isn't `run_until_complete` will wait for the `self._start_supervisor_actor` to finish before returning job_id, which didn't match the comment here that submit will always return immediately ? I think `create_task()` will let asyncio takes care of the execution and return early instead.